packages/savings-vault/src/lib/savings-vault.ts

```diff
@@ -0,0 +1,51 @@
+import type { Address, ContractFactory } from '@phuture/types';
+import { Account } from '@phuture/account';
+import { Erc4626 } from '@phuture/erc-4626';
+import { formatUnits } from 'ethers/lib/utils';
+import { isAddress, Network } from '@phuture/types';
+import {
+	SavingsVault as SavingsVaultContractInterface,
+	SavingsVault__factory,
+} from '../types';
+import {
+	DefaultSavingsVaultViewsAddress,
+	SavingsVaultViews,
+} from './savings-vault-views';
+
+/**
+ * ### SavingsVault Contract
+ */
+export class SavingsVault extends Erc4626<SavingsVaultContractInterface> {
+	/** ### SavingsVaultViews */
+	private _savingsVaultViews: SavingsVaultViews;
+	/**
+	 * ### Creates a new SavingsVault instance
+	 *
+	 * @param account Account to use for interacting with the contract
+	 * @param contract Contract instance or address of the SavingsVault contract
+	 * @param factory Contract factory to use for creating the contract
+	 * @param savingsVaultViews Contract instance or address of the SavingsVaultViews contract
+	 *
+	 * @returns New SavingsVault instance
+	 */
+	constructor(
+		account: Account,
+		contract: Address | SavingsVaultContractInterface,
+		factory: ContractFactory = SavingsVault__factory,
+		savingsVaultViews:
+			| Address
+			| SavingsVaultViews = DefaultSavingsVaultViewsAddress[Network.Mainnet]
+	) {
+		super(account, contract, factory);
+		this._savingsVaultViews = isAddress(savingsVaultViews)
+			? new SavingsVaultViews(account, savingsVaultViews)
+			: savingsVaultViews;
+	}
+
+	public async getAPY(): Promise<string> {
```

Comment:
see our naming examples, 
```suggestion
	public async Apy(): Promise<string> {
```

Reply:
Not sure what you mean. As far as I see, other methods are named like `inactiveAnatomy` or `scaleAmount`

Reply:
yeah, I thinks @jn-lp means that we don't need to use `get` prefix

Reply:
Aha, get it!